packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -3710,10 +3723,14 @@ export class ContainerRuntime
 		// to close current batch.
 		this.flush();
 
-		return {
-			pending: this.pendingStateManager.getLocalState(),
+		const pendingOps = this.pendingStateManager.getLocalState();
+		const pendingState: IPendingRuntimeState = {
+			pending: pendingOps,
 			pendingAttachmentBlobs,
+			// we don't need to save these if we don't have pending ops
+			clientIds: pendingOps ? Array.from(this.stashClientIds) : undefined,
```

Comment:
not something we need to do here, but if there are no ops, there should be no blobs either, so we shouldn't have anything to return, in which case i'd like us to return undefined, and then us that to return undefined at the loader, such that no pending state is returned if there are no local changes.